hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/FinalizeParentRecordHandler.java

```diff
@@ -113,24 +80,33 @@ public void handle(@NonNull final HandleContext context) throws HandleException
         }
 
         /* ------------------------- Hbar changes from transaction including staking rewards ------------------------- */
-        final var hbarChanges = hbarChangesWithStakingRewards(writableAccountStore, readableAccountStore);
+        final var hbarChanges = hbarChangesFrom(writableAccountStore);
+        // any hbar changes listed in child records should not be recorded again in parent record, so deduct them.
+        deductChangesFromChildRecords(hbarChanges, childRecords);
         if (!hbarChanges.isEmpty()) {
             // Save the modified hbar amounts so records can be written
-            recordBuilder.transferList(
-                    TransferList.newBuilder().accountAmounts(hbarChanges).build());
+            recordBuilder.transferList(TransferList.newBuilder()
+                    .accountAmounts(asAccountAmounts(hbarChanges))
+                    .build());
         }
 
         // Declare the top-level token transfer list, which list will include BOTH fungible and non-fungible token
         // transfers
         final ArrayList<TokenTransferList> tokenTransferLists;
 
         // ---------- fungible token transfers
-        final var fungibleChanges = calculateFungibleChanges(writableTokenRelStore, readableTokenRelStore);
-        final var fungibleTokenTransferLists = moldFungibleChanges(fungibleChanges);
+        final var fungibleChanges = fungibleChangesFrom(writableTokenRelStore);
+        // any fungible token changes listed in child records should not be recorded again in parent record, so deduct
+        // them.
+        deductFTChangesFromChildRecords(fungibleChanges, childRecords);
```

Comment:
Because only contract operations can create child records (and no top-level contract operation should have `tokenTransferList`s, I'd say we can remove this method and `deductNftChangesFromChildRecords()` below).

Reply:
Done. Removed. Thanks @tinker-michaelj